src/DynamoManipulation/NodeManipulator.cs

```diff
@@ -172,8 +201,13 @@ protected virtual void MouseDown(object sender, MouseButtonEventArgs mouseButton
         protected virtual void MouseUp(object sender, MouseButtonEventArgs e)
         {
             GizmoInAction = null;
-            //Delete all transient axis line geometry
-            BackgroundPreviewViewModel.DeleteGeometryForIdentifier(RenderDescriptions.AxisLine);
+
+            //Delete all transient graphics for gizmos
+            var gizmos = GetGizmos(false);
+            foreach (var gizmo in gizmos)
+            {
+                gizmo.UpdateGizmoGraphics();
+            }
```

Comment:
Every time the camera changes, the gizmo redraws itself based on the current `Origin` coordinates computed in the getter. This is because the `GetDrawables` logic only depends on axis directions and `Origin`. However the hit test logic actually relies on the cached gizmo geometry (`Axes` and `Planes` properties) for ray intersections. These therefore need to be updated as well since these are used in the hit test logic. I thought we could just update it here in the mouse up event instead of re-evaluating them for every single camera changed event. This design could be improved though.